scripts/xxdi.py

```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/env python3
+# xxdi.py - Pure Python3 implementation of 'xxd -i [input] [output]'
+import sys
+from functools import partial
+
+count = 0
+
+# Convert '/', '$' and '.' to '_',
+# or if this is stdin just use "stdin" as the name.
+argc = len(sys.argv)
+if argc == 1:
+    fd_in = sys.stdin.buffer
+    target_name = 'stdin'
+    fd_out = sys.stdout
+elif argc == 2 or argc == 3:
+    fd_in = open(sys.argv[1], "rb")
+    target_name = sys.argv[1].replace('/','_').replace('$','_').replace('.','_')
+    if argc == 3:
+        fd_out = open(sys.argv[2], "w+")
+    else:
+        fd_out = sys.stdout
+else:
+	print("Too many arguments!", file=sys.stderr)
+	print("Usage: xxdi.py [input] [output]", file=sys.stderr)
+	exit(1)
+
+print("unsigned char %s[] = {" % target_name, file=fd_out)
```

Comment:
Could you make the header
```
#pragma once
#include <stdint.h>
static const uint8_t %s[] = {
```

And also make target_name strip the path, so only the sanitized file name remains?